### PR TITLE
prov/netdir: Fix hanging of protocol in pingpong smoke test

### PR DESCRIPTION
**Pull request by gladkovdmitry17**

### CODE REVIEW

**Comment by gladkovdmitry17** [comment]
Please, don't merge it. I guess, some additional changes are needed. Thank you.

File: prov/netdir/netdir_cq.c [hunk 1]
```diff
@@ -677,10 +678,10 @@ void ofi_nd_unexp_2_read(nd_cq_entry *entry, void *unexpected)
...
 
 	ofi_nd_repack_iovecs(from_iovecs, location_cnt,
-		entry->iov, entry->iov_cnt,
-		new_iovecs, &new_iovecs_count,
-		from_split_map, to_split_map, remote_addr);
-
+			     entry->iov, entry->iov_cnt,
+			     new_iovecs, &new_iovecs_count,
+			     from_split_map, to_split_map, remote_addr);
+	ND_LOG_DEBUG(FI_LOG_EP_DATA, "LOCATIONS Count = %d EntryIOVCount = %d NewIOVCount = %d\n", location_cnt, entry->iov_cnt, new_iovecs_count);
```
Comment (shefty) [inline prov/netdir/netdir_cq.c]: please, for my sanity, limit line lengths to roughly 80 characters :)

File: prov/netdir/netdir_log.h [hunk 2]
```diff
@@ -77,6 +90,103 @@ static inline char *ofi_nd_strerror(DWORD err, HMODULE module)
...
+		OFI_NDERR(ND_INVALID_BUFFER_SIZE, err_str);
+		OFI_NDERR(ND_TOO_MANY_ADDRESSES, err_str);
+		OFI_NDERR(ND_ADDRESS_ALREADY_EXISTS, err_str);
+		OFI_NDERR(ND_CONNECTION_REFUSED, err_str);
+		OFI_NDERR(ND_CONNECTION_INVALID, err_str);
+		OFI_NDERR(ND_CONNECTION_ACTIVE, err_str);
+		OFI_NDERR(ND_HOST_UNREACHABLE, err_str);
+		OFI_NDERR(ND_CONNECTION_ABORTED, err_str);
+		OFI_NDERR(ND_DEVICE_REMOVED, err_str);
+		OFI_NDERR(ND_DISCONNECTED, err_str);
```
Comment (shefty) [inline prov/netdir/netdir_log.h]: outdent OFI_NDERR to align with switch statement

**Comment by shefty** [comment]
@gladkovdmitry17 - based on your last comment, this will not be merged until you reply indicating that everything is ready for another review.

**Comment by gladkovdmitry17** [comment]
@shefty Yes, I guess a couple of days are needed to complete the solution for the PR. Sorry for such long delay, but I didn't have time to work on that

**Comment by gladkovdmitry17** [comment]
@shefty I think everything is ready now and can be reviewed. Thanks!

**Comment by gladkovdmitry17** [comment]
@shefty, I'd like to add correct way to support right sequence for rendezvous protocol in this PR, because NetDir doesn't guarantee ordering of sequence of RMA operations. I think, this PR will be ready tomorrow/ Thanks!

**Comment by shefty** [comment]
I just want to check the status of this PR.  Is this ready for merging?

**Comment by gladkovdmitry17** [comment]
@shefty Unfortunately, I have no time to deal with that right now. But as far as I remember you want to  get ND provider in 1.5 release. If it still so, I can continue with that PR (it's not so much work is remained within that PR)

**Comment by shefty** [comment]
I was just checking the status of this PR.  I'm fine leaving it until it's ready, or closing it and letting you open a new one.

For 1.5, yes, I would like to include ND as part of it.  If you think you can get ND ready within the next 1-2 weeks, please do so.  Otherwise, we'll pull in ND into the 1.6 release, which should be near the end of Q3.  Your choice.

**Comment by gladkovdmitry17** [comment]
@shefty The PR is ready, I checked it with fabtests and seems the provider has an acceptable quality and can be merged. But there is still some minor known issue - the double unloading of dll. I'll spend the next week to looking for proper solution for that.
In the latest commit - The sources files was moved from `/prov/netdir/` to `/prov/netdir/src` directory as it is in other providers.

**Comment by shefty** [comment]
Thanks for the update - merged.

**Comment by gladkovdmitry17** [comment]
@shefty Thanks too!

**Pull request merged**